sky/skylet/providers/azure/config.py

```diff
@@ -50,12 +49,15 @@ def _configure_resource_group(config):
     # TODO: look at availability sets
     # https://docs.microsoft.com/en-us/azure/virtual-machines/windows/tutorial-availability-sets
     subscription_id = config["provider"].get("subscription_id")
-    if subscription_id is None:
-        subscription_id = get_cli_profile().get_subscription_id()
     # Increase the timeout to fix the Azure get-access-token (used by ray azure
     # node_provider) timeout issue.
     # Tracked in https://github.com/Azure/azure-cli/issues/20404#issuecomment-1249575110
     credentials = AzureCliCredential(process_timeout=30)
+
+    if subscription_id is None:
+        subscription_client = SubscriptionClient(credentials)
```

Comment:
Is there a reason we does not use `json.loads(_run_output('az account show -o json'))` here? Or use this method in `sky/adaptors/azure.py`? Why are we using different method here, and what is the difference between them?